indico/modules/events/controllers/admin.py

```diff
@@ -121,6 +122,21 @@ def _process(self):
         return jsonify_form(form)
 
 
+class RHUpdateEventKeywords(RHAdminBase):
+    """Update event keywords."""
+
+    def _process(self):
+        form = EventKeywordsForm()
+        keywords = None
+        if form.validate_on_submit():
+            keywords = sorted({kw.lower() for kw in form.data.get('keywords', [])})
+            misc_settings.set('allowed_keywords', keywords)
+            flash(_('Allowed keywords have been saved'), 'success')
+        if not keywords:
+            keywords = misc_settings.get('allowed_keywords')
```

Comment:
I think this can be done unconditionally

Reply:
Not really, if you just set the keywords on the first `if` the keywords are already calculated and there's no need to make another call.

Reply:
I usually follow "keep it simple" and "no premature optimization". ;)
You are of course absolutely right that it's not *necessary* to get the keywords again.

Anyway, I just realized that this whole discussion is pointless because this RH is missing one important thing: `return redirect(url_for('.event_keywords'))` after the `flash()`. After a POST that modified data it's recommended to redirect back to GET (so F5 doesn't trigger the "submit again" prompt and more importantly doesn't repeat the modification).

Reply:
Totally right. This way it does not send another POST if refreshing the page.